packages/editor/src/components/document-bar/index.js

```diff
@@ -48,39 +48,56 @@ const icons = {
 };
 
 export default function DocumentBar() {
-	const { isEditingTemplate, templateId, postType, postId } = useSelect(
-		( select ) => {
-			const {
-				getRenderingMode,
-				getCurrentTemplateId,
-				getCurrentPostId,
-				getCurrentPostType,
-			} = select( editorStore );
-			const _templateId = getCurrentTemplateId();
-			return {
-				isEditingTemplate:
-					!! _templateId && getRenderingMode() === 'template-only',
-				templateId: _templateId,
-				postType: getCurrentPostType(),
-				postId: getCurrentPostId(),
-			};
-		},
-		[]
-	);
+	const {
+		isEditingTemplate,
+		templateId,
+		postType,
+		postId,
+		hasHistory,
+		goBack,
+	} = useSelect( ( select ) => {
+		const {
+			getRenderingMode,
+			getCurrentTemplateId,
+			getCurrentPostId,
+			getCurrentPostType,
+			getEditorSettings,
+		} = select( editorStore );
+		const _templateId = getCurrentTemplateId();
+		const back = getEditorSettings().goBack;
+		return {
+			isEditingTemplate:
+				!! _templateId && getRenderingMode() === 'template-only',
+			templateId: _templateId,
+			postType: getCurrentPostType(),
+			postId: getCurrentPostId(),
```

Comment:
I wonder if we need to consider how these APIs work with the history, as there's a lot of overlap. 🤔

Reply:
I don't think we should but in the site editor for instance, calling `onSelectPost` should actually do nothing more than just `history.navigate` and potentially, it should store the entity somewhere to be able to do `goBack` (actually not even sure it should store the entity as it can check from the "history" object maybe).

In other words, it's the "history" that will control these props as they get passed to `EditorProvider`. In the post editor it might not be needed unless we decide to add "url changes" there.